twill-core/src/main/java/org/apache/twill/internal/json/TwillRunResourcesCodec.java

```diff
@@ -66,13 +67,14 @@ public JsonElement serialize(TwillRunResources src, Type typeOfSrc, JsonSerializ
   public TwillRunResources deserialize(JsonElement json, Type typeOfT,
                                            JsonDeserializationContext context) throws JsonParseException {
     JsonObject jsonObj = json.getAsJsonObject();
+    Map<String, LogEntry.Level> logLevels =
+      context.deserialize(jsonObj.get("logLevels"), new TypeToken<Map<String, LogEntry.Level>>() { }.getType());
```

Comment:
Since constants are defined, we should use those constants in deserialization as well.